src/components/selectionBoxLayer.ts

```diff
@@ -111,6 +131,84 @@ export module Components {
     public fixedHeight() {
       return true;
     }
+
+    /**
+     * Gets the x scale for this SelectionBoxLayer.
+     */
+    public xScale<D extends number | { valueOf(): number }>(): QuantitativeScale<D>;
+    /**
+     * Sets the x scale for this SelectionBoxLayer.
+     * 
+     * @returns {SelectionBoxLayer} The calling SelectionBoxLayer.
+     */
+    public xScale<D extends number | { valueOf(): number }>(xScale: QuantitativeScale<D>): SelectionBoxLayer;
+    public xScale<D extends number | { valueOf(): number }>(xScale?: QuantitativeScale<D>): any {
+      if (xScale == null) {
+        return this._xScale;
+      }
+      if (this._xScale != null) {
+        this._xScale.offUpdate(this._adjustBoundsCallback);
+      }
+      this._xScale = xScale;
+      this._xScale.onUpdate(this._adjustBoundsCallback);
+      this._bindBoxDataValues();
+      return this;
+    }
+
+    /**
+     * Gets the y scale for this SelectionBoxLayer.
+     */
+    public yScale<D extends number | { valueOf(): number }>(): QuantitativeScale<D>;
+    /**
+     * Sets the y scale for this SelectionBoxLayer.
+     * 
+     * @returns {SelectionBoxLayer} The calling SelectionBoxLayer.
+     */
+    public yScale<D extends number | { valueOf(): number }>(yScale: QuantitativeScale<D>): SelectionBoxLayer;
+    public yScale<D extends number | { valueOf(): number }>(yScale?: QuantitativeScale<D>): any {
+      if (yScale == null) {
+        return this._yScale;
+      }
+      if (this._yScale != null) {
+        this._yScale.offUpdate(this._adjustBoundsCallback);
+      }
+      this._yScale = yScale;
+      this._yScale.onUpdate(this._adjustBoundsCallback);
+      this._bindBoxDataValues();
+      return this;
+    }
+
+    /**
+     * Gets the data values for the Bounds of the box.
+     *
+     * An undefined value will be returned if a scale is not backing a given edge, i.e.
+     * no xScale is backing the left / right edges of the box.
+     */
+    public dataBounds() {
```

Comment:
Further, what gets returned for an `XDragBoxLayer`? Would I expect `null` for the Y data values?